### PR TITLE
Refactor cfn templates

### PR DESCRIPTION
**Pull request by rupertbates**
## Why are you doing this?
The way we currently build our Cloud formation templates is a getting a bit unwieldy now that we have a lot of lambdas and some nested templates. 

Using a proper templating language (Mustache) makes it much easier to understand what is going on and reduces the amount of boilerplate. I have switched from Python to Node for the execution environment as it has better support for working with Mustache templates.

This PR just replaces the old system with the new one. After merging this I will move on to adding the new templates needed for the Digital Bundle.

There will also need to be some additional changes to get the new system to build in TeamCity, which I will probably add in to this PR.

[**Trello Card**](https://trello.com/c/dWpYddPv/792-new-cloudformation-templates-for-support-workers-digital-bundle)

### CODE REVIEW

**Comment by JustinPinner** [comment]
Took me a few re-reads and much frowning to parse this into wetware, but I think I'm beginning to understand. 👍 from me (but please seek others, too!)

File: cloud-formation/src/templates/lambda.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+  {{name}}:
+    Type: "AWS::Lambda::Function"
+    Description: "{{description}}"
+    Properties:
+      {{> environmentVariables}}
+
+      FunctionName: !Sub ${Stack}-ContributionCompletedLambda-${Stage}
```
Comment (davidfurey) [inline cloud-formation/src/templates/lambda.yaml]: Should the references to "CreatePaymentMethodLambda" not be `{{name}}` or similar?

File: cloud-formation/src/cfn.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (davidfurey) [inline cloud-formation/src/cfn.json]: Could this file be yaml?

File: cloud-formation/src/templates/lambda.yaml (hunk 1 above)
Comment (rupertbates) [inline cloud-formation/src/templates/lambda.yaml]: 👍 Good spot

File: cloud-formation/src/templates/lambda.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+  {{name}}:
+    Type: "AWS::Lambda::Function"
+    Description: "{{description}}"
+    Properties:
+      {{> environmentVariables}}
+
+      FunctionName: !Sub ${Stack}-{{name}}-${Stage}
+      Handler: "com.gu.support.workers.lambdas.ContributionCompleted::handleRequest"
```
Comment (davidfurey) [inline cloud-formation/src/templates/lambda.yaml]: I think `ContributionCompleted` shouldn't be hard-coded here

Comment (rupertbates) [inline cloud-formation/src/templates/lambda.yaml]: It's not any more, not sure why you are seeing it like that.

Comment (rupertbates) [inline cloud-formation/src/templates/lambda.yaml]: Ok fixed now

**Pull request merged**